shared/chains.py

```diff
@@ -229,14 +229,14 @@ def render_address(cls, script):
         if ll == 23 and script[0:2] == b'\xA9\x14' and script[22] == 0x87:
             return ngu.codecs.b58_encode(cls.b58_script + script[2:2+20])
 
-        # P2WPKH
-        if ll == 22 and script[0:2] == b'\x00\x14':
-            return ngu.codecs.segwit_encode(cls.bech32_hrp, 0, script[2:])
-
-        # P2WSH, P2TR and later
-        if ll == 34 and script[0] <= 16 and script[1] == 0x20:
+        # segwit v0 (P2WPKH, P2WSH)
+        if script[0] == 0 and script[1] in (0x14, 0x20) and ll in (22, 34):
             return ngu.codecs.segwit_encode(cls.bech32_hrp, script[0], script[2:])
 
+        # segwit v1 (P2TR) and later segwit version
+        if ll == 34 and 0x51 <= script[0] <= 0x60 and script[1] == 0x20:
```

Comment:
add `()` around `0x51 <= script[0] <= 0x60` for readability

Reply:
used OP_N instead for even better readability